kernel/comps/virtio/src/device/network/device.rs

```diff
@@ -170,29 +191,90 @@ fn queue_to_network_error(err: QueueError) -> VirtioNetError {
     match err {
         QueueError::NotReady => VirtioNetError::NotReady,
         QueueError::WrongToken => VirtioNetError::WrongToken,
+        QueueError::BufferTooSmall => VirtioNetError::Busy,
         _ => VirtioNetError::Unknown,
     }
 }
 
+fn init_caps(features: &NetworkFeatures, config: &VirtioNetConfig) -> DeviceCapabilities {
+    let mut caps = DeviceCapabilities::default();
+
+    caps.max_burst_size = None;
+    caps.medium = Medium::Ethernet;
+
+    if features.contains(NetworkFeatures::VIRTIO_NET_F_MTU) {
+        // If `VIRTIO_NET_F_MTU` is negotiated, the MTU is decided by the device.
+        caps.max_transmission_unit = config.mtu as usize;
+    } else {
+        // Linux does not has a specified MTU for virtio-net.
+        // Linux only requires the MTU between 68-65535.
+        // The exact value can be decided dynamically.
+        // <https://elixir.bootlin.com/linux/v6.0.9/source/drivers/net/virtio_net.c#L3660>.
+
+        if features.contains(NetworkFeatures::VIRTIO_NET_F_GUEST_TSO4)
+            || features.contains(NetworkFeatures::VIRTIO_NET_F_GUEST_TSO6)
+            || features.contains(NetworkFeatures::VIRTIO_NET_F_GUEST_UFO)
```

Comment:
This seems pretty wired to me and I don't think this is correct.

After checking the virtio-net spec, I've found that to use these features, we need to properly initialize the `gso_type` field in the virtio-net header:
```c
struct virtio_net_hdr {
#define VIRTIO_NET_HDR_F_NEEDS_CSUM 1
#define VIRTIO_NET_HDR_F_DATA_VALID 2
#define VIRTIO_NET_HDR_F_RSC_INFO 4
    u8 flags;
#define VIRTIO_NET_HDR_GSO_NONE 0
#define VIRTIO_NET_HDR_GSO_TCPV4 1
#define VIRTIO_NET_HDR_GSO_UDP 3
#define VIRTIO_NET_HDR_GSO_TCPV6 4
#define VIRTIO_NET_HDR_GSO_UDP_L4 5
#define VIRTIO_NET_HDR_GSO_ECN 0x80
    u8 gso_type;
    // ...
}
```
However, we only use `VirtioNetHdr::default()`, so the features mentioned here never work, I don't think it makes any difference to us whether these features are negotiated or not.

Reply:
Yes, actually these features are not supported and this branch will not be reached. I use a `todo!` here,